src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -668,23 +668,34 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
                 src = "-" + src;
                 last_unary_plus = true;
                 last_binary_plus = false;
-                return;
+            } else if (x.m_op == ASR::unaryopType::Invert) {
+                src = "~" + src;
+                last_unary_plus = false;
+                last_binary_plus = false;
+            } else if (x.m_op == ASR::unaryopType::Not) {
+                src = "!" + src;
+                last_unary_plus = false;
+                last_binary_plus = false;
             } else {
                 throw CodeGenError("Unary type not implemented yet for Integer");
             }
+            return;
         } else if (x.m_type->type == ASR::ttypeType::Real) {
             if (x.m_op == ASR::unaryopType::UAdd) {
                 // src = src;
                 last_unary_plus = false;
-                return;
             } else if (x.m_op == ASR::unaryopType::USub) {
                 src = "-" + src;
                 last_unary_plus = true;
                 last_binary_plus = false;
-                return;
+            } else if (x.m_op == ASR::unaryopType::Not) {
+                src = "!" + src;
+                last_unary_plus = false;
+                last_binary_plus = false;
```

Comment:
Btw, these `last_unary_plus` do not work well. I did it like this because I thought it would work, but it doesn't work. The way to implement this is using this approach: 

* https://gitlab.com/lfortran/lfortran/-/blob/127b40f5c6c074a763b104e86fe599c3e77ef798/src/lfortran/ast_to_src.cpp#L114

* https://gitlab.com/lfortran/lfortran/-/blob/127b40f5c6c074a763b104e86fe599c3e77ef798/src/lfortran/ast_to_src.cpp#L3367

* https://gitlab.com/lfortran/lfortran/-/blob/127b40f5c6c074a763b104e86fe599c3e77ef798/src/lfortran/ast_to_src.cpp#L3371